entity-service-change-event-api/build.gradle.kts

```diff
@@ -0,0 +1,68 @@
+import com.google.protobuf.gradle.generateProtoTasks
+import com.google.protobuf.gradle.id
+import com.google.protobuf.gradle.ofSourceSet
+import com.google.protobuf.gradle.plugins
+import com.google.protobuf.gradle.protobuf
+import com.google.protobuf.gradle.protoc
+
+plugins {
+  `java-library`
+  id("com.google.protobuf") version "0.8.15"
+  id("org.hypertrace.publish-plugin")
+}
+
+val generateLocalGoGrpcFiles = false
+
+protobuf {
+  protoc {
+    artifact = "com.google.protobuf:protoc:3.15.6"
+  }
+  plugins {
+    // Optional: an artifact spec for a protoc plugin, with "grpc" as
+    // the identifier, which can be referred to in the "plugins"
+    // container of the "generateProtoTasks" closure.
+    id("grpc_java") {
+      artifact = "io.grpc:protoc-gen-grpc-java:1.40.1"
+    }
+
+    if (generateLocalGoGrpcFiles) {
+      id("grpc_go") {
+        path = "<go-path>/bin/protoc-gen-go"
+      }
+    }
+  }
+  generateProtoTasks {
+    ofSourceSet("main").forEach {
+      it.plugins {
+        // Apply the "grpc" plugin whose spec is defined above, without options.
+        id("grpc_java")
+
+        if (generateLocalGoGrpcFiles) {
+          id("grpc_go")
+        }
+      }
+      it.builtins {
+        java
+
+        if (generateLocalGoGrpcFiles) {
+          id("go")
+        }
+      }
+    }
+  }
+}
+
+sourceSets {
+  main {
+    java {
+      srcDirs("src/main/java", "build/generated/source/proto/main/java", "build/generated/source/proto/main/grpc_java")
+    }
+  }
+}
+
+dependencies {
+  api("io.grpc:grpc-protobuf:1.40.1")
+  api("io.grpc:grpc-stub:1.40.1")
+  api("javax.annotation:javax.annotation-api:1.3.2")
+  protobuf(project(":entity-service-api"))
```

Comment:
hmm - we really don't want proto packages depending on each other, which I guess is the problem wiith spliitting this out.

At a minimum, I think we'd change this to an API dependency, but that requires versioning it. Let's discuss this part.

Reply:
just curious what is the problem if this depends on the other ? is it due to the possibility that somebody might import a certain version of entity-service-api and might have a different version of  entity-service-change-event-api ? I guess if both are on the same version then it shouldn't be a problem